crates/evm/core/src/decode.rs

```diff
@@ -124,11 +124,28 @@ pub fn maybe_decode_revert(
 }
 
 fn trimmed_hex(s: &[u8]) -> String {
-    let s = hex::encode(s);
-    let n = 32 * 2;
+    let n = 32;
     if s.len() <= n {
-        s
+        hex::encode(s)
     } else {
-        format!("{}…{} ({} bytes)", &s[..n / 2], &s[s.len() - n / 2..], s.len())
+        format!(
+            "{}…{} ({} bytes)",
+            &hex::encode(&s[..n / 2]),
+            &hex::encode(&s[s.len() - n / 2..]),
```

Comment:
hmm, not instantly obvious to me what the difference is

this now operates on the input and not the hex encoded input while also using 32 bytes instead of 64, but the slicing operations are the same?

so this is now only encodes the necessary slices? is that the change?

Reply:
the key change is that `s.len()` (the third argument of the `format!` invocation) now operates on the input slice, so it will print the correct length (where previously a shadowing binding for `s` was introduced so `s.len()` operated on the hex string, doubling the length).

the original bug is fixable with a smaller diff (just putting `s.len() / 2` in the 3rd argument), I'm happy to do that instead if that's preferred

Reply:
> so this is now only encodes the necessary slices? is that the change?

but yes, also only encodes the necessary slices -- technically that's an orthogonal change, but I thought the code looked a bit funny when all the operations were scaled back by 2

Reply:
it was less calls to `hex::encode`, but it doesn't matter